controllers/machineset_controller.go

```diff
@@ -249,6 +248,9 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 	}
 
 	syncErr := r.syncReplicas(ctx, machineSet, filteredMachines)
+	if syncErr != nil {
+		return ctrl.Result{}, errors.Wrapf(syncErr, "failed to sync MachineSet replicas")
+	}
```

Comment:
We should still check this error after `calculateStatus` or we might display outdated information to users in the MS' status